File: GeneratedEntityReaderClasses/Components/StatusEffectDataComponent.cs
namespace GeneratedNoitaClasses;
public class StatusEffectDataComponent : NoitaComponentBase,  iNoitaType<StatusEffectDataComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaVector<NoitaFloat> stain_effects { get; set; } = new();
	public NoitaVector<NoitaInt> stain_effect_cooldowns { get; set; } = new();
	public NoitaVector<NoitaFloat> effects_previous { get; set; } = new();
	public NoitaVector<NoitaFloat> ingestion_effects { get; set; } = new();
	public NoitaVector<NoitaInt> ingestion_effect_causes { get; set; } = new();
	public NoitaVector<NoitaInt> ingestion_effect_causes_many { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		stain_effects.Read(s);
		stain_effect_cooldowns.Read(s);
		effects_previous.Read(s);
		ingestion_effects.Read(s);
		ingestion_effect_causes.Read(s);
		ingestion_effect_causes_many.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		stain_effects.Write(s);
		stain_effect_cooldowns.Write(s);
		effects_previous.Write(s);
		ingestion_effects.Write(s);
		ingestion_effect_causes.Write(s);
		ingestion_effect_causes_many.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/StatusEffectDataComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class StatusEffectDataComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    stain_effects:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaFloat))
    stain_effect_cooldowns:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaInt))
    effects_previous:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaFloat))
    ingestion_effects:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaFloat))
    ingestion_effect_causes:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaInt))
    ingestion_effect_causes_many:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaInt))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.stain_effects.Read(s)
        self.stain_effect_cooldowns.Read(s)
        self.effects_previous.Read(s)
        self.ingestion_effects.Read(s)
        self.ingestion_effect_causes.Read(s)
        self.ingestion_effect_causes_many.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.stain_effects.Write(s)
        self.stain_effect_cooldowns.Write(s)
        self.effects_previous.Write(s)
        self.ingestion_effects.Write(s)
        self.ingestion_effect_causes.Write(s)
        self.ingestion_effect_causes_many.Write(s)

